.github/workflows/nightly_ondemand_whl.yml

```diff
@@ -0,0 +1,335 @@
+  cancel-in-progress: true
+
+jobs:
+  Linux-Nightly-Ondemand-UT-WHL-Tests:
+    if: github.event_name == 'schedule' || ${{ inputs.ut_suite }}
+    uses: ./.github/workflows/_linux_ut.yml
+    with: 
+      ut: ${{ github.event_name == 'schedule' && 'op_example,op_extended,op_ut,torch_xpu' || inputs.ut }}
+      pytorch: ${{ github.event_name == 'schedule' && 'main' || inputs.pytorch }}
+      python: ${{ github.event_name == 'schedule' && '3.10' || inputs.python }}
+      test_type: whl
+      runner: linux.idc.xpu
+
+  Linux-Nightly-Ondemand-E2E-WHL-Tests:
+    runs-on: pvc_e2e
+    # Don't run on forked repos
+    if: github.repository_owner == 'intel'
+    timeout-minutes: 3600
+    env:
+      pytorch: ${{ github.event_name == 'schedule' && 'main' || inputs.pytorch }}
+      ut: ${{ github.event_name == 'schedule' && 'op_example,op_extended,op_ut,torch_xpu' || inputs.ut }}
+      python: ${{ github.event_name == 'schedule' && '3.10' || inputs.python }}
+    outputs:
+      TORCH_BRANCH_ID: ${{ steps.pinned.outputs.TORCH_BRANCH_ID }}
+      TORCH_COMMIT_ID: ${{ steps.pinned.outputs.TORCH_COMMIT_ID }}
+      DRIVER_VERSION: ${{ steps.pinned.outputs.DRIVER_VERSION }}
+      BUNDLE_VERSION: ${{ steps.pinned.outputs.BUNDLE_VERSION }}
+      OS_PRETTY_NAME: ${{ steps.pinned.outputs.OS_PRETTY_NAME }}
+      GCC_VERSION: ${{ steps.pinned.outputs.GCC_VERSION }}
+      TORCHBENCH_COMMIT_ID: ${{ steps.pinned.outputs.TORCHBENCH_COMMIT_ID }}
+      TORCHVISION_COMMIT_ID: ${{ steps.pinned.outputs.TORCHVISION_COMMIT_ID }}
+      TORCHAUDIO_COMMIT_ID: ${{ steps.pinned.outputs.TORCHAUDIO_COMMIT_ID }}
+      TRANSFORMERS_VERSION: ${{ steps.pinned.outputs.TRANSFORMERS_VERSION }}
+      TIMM_COMMIT_ID: ${{ steps.pinned.outputs.TIMM_COMMIT_ID }}
+      TRITON_COMMIT_ID: ${{ steps.pinned.outputs.TRITON_COMMIT_ID }}
+      TIMEOUT_MODELS: ${{ steps.summary.outputs.TIMEOUT_MODELS }}
+    steps:
+      - name: Checkout torch-xpu-ops
+        uses: actions/checkout@v4
+      - name: Prepare Conda ENV
+        run: |
+          which conda && conda clean -ay
+          conda remove --all -y -n e2e_ci || rm -rf $(dirname ${CONDA_EXE})/../envs/e2e_ci
+          conda create -n e2e_ci python=${{ env.python }} cmake ninja -y
+          source activate e2e_ci
+          pip install mkl-static mkl-include
+          pip install pandas scipy tqdm
+      - name: Prepare Stock Pytorch
+        run: |
+          pwd
+          cd ../ && rm -rf pytorch
+          source activate e2e_ci
+          git clone https://github.com/pytorch/pytorch pytorch
+          cd pytorch && git checkout ${{ env.pytorch }} 
+          # apply PRs for stock pytorch
+          pip install requests
+          python ../torch-xpu-ops/.github/scripts/apply_torch_pr.py
+          git status && git show -s
+      - name: Identify pinned versions
```

Comment:
may need to consider how to get accurate sw info in this whl test workflow, eg. torch/torchvision/audio version, we may get it from pip list after installation

Reply:
Sure, torch/torchvision/torchaudio/triton versions are from pip list